atlasdb-commons/src/main/java/com/palantir/common/remoting/ServiceNotAvailableException.java

```diff
@@ -15,46 +15,78 @@
  */
 package com.palantir.common.remoting;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.net.HostAndPort;
+import com.google.errorprone.annotations.CompileTimeConstant;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
 
 /**
- * If a server is shutting down or cannot respond to a call for another reson this exception may be thrown.
+ * If a server is shutting down or cannot respond to a call for another reason this exception may be thrown.
  * This exception indicates to the caller that this call should be retried on another server that is available.
  *
  * @author carrino
  */
-public class ServiceNotAvailableException extends RuntimeException {
+public class ServiceNotAvailableException extends RuntimeException implements SafeLoggable {
```

Comment:
We have quite a few `extends RuntimeExceptions` that we could make SafeLoggable. Any nacks on me just changing those in a separate PR, where appropriate?

Reply:
![Screenshot 2023-10-16 at 16 55 47](https://github.com/palantir/atlasdb/assets/1812764/3dc7f87f-7ed8-4f2c-8db1-2dbbbbc76976)

There's also a few exceptions (like TableMappingNotFoundException worth doing too).

I reckon it would take me ~30 mins for the lot (not counting review time!)

Reply:
yeah this sounds good. Definitely separate though.